firebase/FirestoreCtrl.ts

```diff
@@ -282,9 +292,83 @@ export default class FirestoreCtrl {
     }
   }
 
+  /**
+   * Retrieves all groups assigned to a specific user.
+   *
+   * @param uid The UID of the user whose groups are to be fetched.
+   * @returns A promise that resolves to an array of groups.
+   */
+  async getGroupsByUserId(uid: string): Promise<DBGroup[]> {
+    try {
+      const userRef = doc(firestore, "users", uid);
+      const userDoc = await getDoc(userRef);
+
+      const userData = userDoc.data() as DBUser;
+
+      console.log("userGroups [" + uid + "]", userData.groups);
+
+      // Retrieve all groups using the group IDs
+      const groupsRef = collection(firestore, "groups");
+
+      const q = query(groupsRef, where(documentId(), "in", userData.groups));
+
+      const groupSnapshots = await getDocs(q);
+
+      // Map the results into an array of DBGroup
+      const dbGroups: DBGroup[] = groupSnapshots.docs.map(
+        (doc) =>
+          ({
+            group_id: doc.id,
+            ...doc.data(),
+          }) as DBGroup,
+      );
+
+      return dbGroups;
+    } catch (error) {
+      console.error("Error getting groups by user ID: ", error);
+      throw error;
+    }
+  }
+
+  /**
+   * Retrieves all groups assigned to a specific user.
+   *
+   * @param uid The UID of the user whose groups are to be fetched.
+   * @returns A promise that resolves to an array of groups.
+   */
+  async getUsersInGroup(gid: string): Promise<DBUser[]> {
+    try {
+      const groupRef = doc(firestore, "groups", gid);
+      const groupDoc = await getDoc(groupRef);
+
+      const userData = groupDoc.data() as DBGroup;
+
+      // Retrieve all users using the user IDs
+      const usersRef = collection(firestore, "users");
+      const q = query(usersRef, where("uid", "in", userData.members));
```

Comment:
Be aware of potential limited queries, could be fixed in a future PR.

Reply:
I do not use this function for now but will change it in next implementation thank you